### PR TITLE
Re SI-6688

### PR DESCRIPTION
**Pull request by Johnlon**
CompileClient now connects by default to the "localhost/127.0.0.1" interface rather than the public external facing local interface.

### CODE REVIEW

**Comment by paulp** [comment]
This commit requires some elaboration. Among other things, it is hard to believe that hardcoding the IPv4 loopback address is optimal, regardless of how frequently that particular magic number changes.

File: src/compiler/scala/tools/nsc/io/Socket.scala [hunk 1]
```diff
@@ -35,7 +35,7 @@ object Socket {
 
   def newIPv4Server(port: Int = 0)        = new Box(() => preferringIPv4(new ServerSocket(0)))
   def newServer(port: Int = 0)            = new Box(() => new ServerSocket(0))
-  def localhost(port: Int)                = apply(InetAddress.getLocalHost(), port)
```
Comment (paulp) [inline src/compiler/scala/tools/nsc/io/Socket.scala:38]: I believe it should be InetAddress.getByName(null).

**Comment by jsuereth** [comment]
Three things:

(1) you'll have to give me time to test this on several platforms.  We're lacking in FSC tests.
(2) Have you signed the scala CLA?
(3) Did you want to target 2.10.1?

**Comment by Johnlon** [comment]
1) Sure
2) Not yet - do I have to use snail mail - can the change be made without it? 
3) I need it in 2.9.*
Thanks loads.

**Comment by adriaanm** [comment]
2) you can email a pdf to danielle.chamberlain@epfl.ch (see http://www.scala-lang.org/sites/default/files/contributor_agreement.pdf)
3) in that case please open a new pull request (that address the reviewer feedback) on the 2.9.x branch

**Comment by adriaanm** [comment]
Thanks!

**Pull request closed without merging**